### PR TITLE
CI-CD Updates and Repository PR Checks

### PR DESCRIPTION
**Pull request by Skptak**
Minor updates to some of the CI-CD files to remove trailing whitespace, and fix formatting issues
Use checkout@v3 in all places instead of v2
Fixes to the README for the repo
Add in a new test that runs on PRs into this repo so that potential changes are tested against known consumers of this repository

### CODE REVIEW

**Comment by Skptak** [comment]
Example of how this looks for a failed lcov report can be found [here](https://github.com/Skptak/ota-for-aws-iot-embedded-sdk/actions/runs/5861951105) 
Example of how this looks for a failed formatting patch diff can be found [here](https://github.com/Skptak/ota-for-aws-iot-embedded-sdk/actions/runs/5861955589):

**Comment by Skptak** [comment]
Workflow step running against all my forks can be seen [here](https://github.com/Skptak/CI-CD-Github-Actions/actions/runs/5862098848)

**Comment by Skptak** [comment]
Final commit showing the PR checks passing can be found [here](https://github.com/FreeRTOS/CI-CD-Github-Actions/actions/runs/5862324925)
Going to swap it over to the versions at github.com/aws and github.com/FreeRTOS at this point for the sake of the PR

File: .github/workflows/pr_checks.yml [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+           #  exclude-files: .cSpellWords.txt,
+           #  run-complexity: false,
+           #  org: FreeRTOS,
+           #},
+           #{
+           #  repository: FreeRTOS-Plus-TCP,
+           #  exclude-files: .cSpellWords.txt,
+           #  run-complexity: false,
+           #  org: FreeRTOS,
+           #},
```
Comment (rawalexe) [inline .github/workflows/pr_checks.yml:165]: commented code??

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -7,357 +7,352 @@ on:
...
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
+
+      - uses: actions/checkout@v3
         with:
-          repository: FreeRTOS/coreMQTT
+          repository: skptak/coreMQTT
```
Comment (rawalexe) [inline .github/workflows/test.yml:65]: why are you targeting your personal fork?

File: executable-monitor/executable-monitor.py [hunk 3]
```diff
@@ -7,16 +7,34 @@
...
+stdout_logging_handler = logging.StreamHandler(sys.stdout)
+stdout_logging_handler.setLevel(logging.DEBUG)
+stdout_logging_formatter = logging.Formatter('%(asctime)s - %(levelname)s - %(message)s')
+stdout_logging_handler.setFormatter(stdout_logging_formatter)
+logging.getLogger().addHandler(stdout_logging_handler)
+
 def runAndMonitor(args):
     # Set up logging
-    logging.getLogger().setLevel(logging.NOTSET)
+    #logging.getLogger().setLevel(logging.NOTSET)
```
Comment (rawalexe) [inline executable-monitor/executable-monitor.py:28]: code commenting...

File: .github/workflows/pr_checks.yml (hunk 1 above)
Comment (Skptak) [reply .github/workflows/pr_checks.yml:165]: Showing how these repos can be included when all work for them is done

File: executable-monitor/executable-monitor.py [hunk 4]
```diff
@@ -7,16 +7,34 @@
...
     # Add stdout handler to logging
-    stdout_logging_handler = logging.StreamHandler(sys.stdout)
-    stdout_logging_handler.setLevel(logging.DEBUG)
-    stdout_logging_formatter = logging.Formatter('%(asctime)s - %(levelname)s - %(message)s')
-    stdout_logging_handler.setFormatter(stdout_logging_formatter)
-    logging.getLogger().addHandler(stdout_logging_handler)
+    #stdout_logging_handler = logging.StreamHandler(sys.stdout)
+    #stdout_logging_handler.setLevel(logging.DEBUG)
+    #stdout_logging_formatter = logging.Formatter('%(asctime)s - %(levelname)s - %(message)s')
+    #stdout_logging_handler.setFormatter(stdout_logging_formatter)
```
[the remark this answers is not in the document]
Comment (Skptak) [reply executable-monitor/executable-monitor.py]: Code removed

**Review by n9wxu — APPROVED** [review]

**Review by bradleysmith23 — DISMISSED** [review]

File: sbom-generator/sbom_utils.py [hunk 5]
```diff
@@ -28,8 +28,8 @@ def file_writer(output, filepath: str, sha1: str, license: str, copyright='NOASS
     output.write('FileCopyrightText: '+ copyright + '\n')
     output.write('FileComment: '+ comment + '\n')
     output.write('\n')
-    
-def pacakge_writer(output, packageName: str, version: str, url: str, license: str, ver_code: str, file_analyzed=True, 
+
+def pacakge_writer(output, packageName: str, version: str, url: str, license: str, ver_code: str, file_analyzed=True,
```
Comment (A-Zaba) [inline sbom-generator/sbom_utils.py:32]: NIT: Spelling -- package_writer

**Review by A-Zaba — APPROVED** [review]

**Review by A-Zaba — APPROVED** [review]

**Review by bradleysmith23 — APPROVED** [review]

**Pull request merged**